yktrace/src/lib.rs

```diff
@@ -12,6 +13,15 @@ mod hwt;
 pub use errors::InvalidTraceError;
 pub use hwt::mapper::BlockMap;
 
+thread_local! {
+    // When `Some` contains the `ThreadTracer` for the current thread. When `None`, the current
+    // thread is not being traced.
+    //
+    // We hide the `ThreadTracer` in a thread local (rather than returning it to the consumer of
+    // yk). This ensures that the `ThreadTracer` itself cannot appear in traces.
+    pub static THREAD_TRACER: RefCell<Option<ThreadTracer>> = RefCell::new(None);
```

Comment:
There is a nightly feature that allows writing

```suggestion
    pub static THREAD_TRACER: RefCell<Option<ThreadTracer>> = const { RefCell::new(None) };
```

This makes TLS accesses cheaper as it doesn't have to check if it needs to lazily initialize it every time, instead evaluating the initializer at compile time and storing it's result in the `.tdata` section which libc copies to form the TLS storage for a thread.

Reply:
Thanks! I've implemented that in eac6196.